state/consumer.go

```diff
@@ -47,7 +53,16 @@ func (k *ConsumersCollection) Add(consumer Consumer) error {
 	if !utils.Empty(consumer.Username) {
 		searchBy = append(searchBy, *consumer.Username)
 	}
-	_, err := getConsumer(txn, searchBy...)
+	_, err := getConsumer(txn, []string{"Username", "id"}, searchBy...)
```

Comment:
I might be wrong here but why can't we inject `custom_id` into `searchBy` and then add `CustomID` to the indices param?